rpcs3/rpcs3qt/syntax_highlighter.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <QSyntaxHighlighter>
+#include <QRegularExpression>
+
+class syntax_highlighter : public QSyntaxHighlighter
+{
+	Q_OBJECT
+
+	enum BlockState
+	{
+		ENDED_OUTSIDE_COMMENT,
+		ENDED_INSIDE_COMMENT
+	};
+
+public:
+	syntax_highlighter(QTextDocument *parent = 0);
+
+	struct SimpleRule
+	{
+		QVector<QRegularExpression> expressions;
+		QTextCharFormat format;
+	};
+	struct MultiRule
+	{
+		QRegularExpression expression;
+		QVector<QTextCharFormat> formats;
+	};
+	struct CommentRule
+	{
+		QRegularExpression start_expression;
+		QRegularExpression end_expression;
+		QTextCharFormat format;
+		bool multi_line = false;
+	};
+
+	/** Add a simple highlighting rule that applies the given format to all given expressions */
+	void AddSimpleRule(SimpleRule& rule);
```

Comment:
Where is this called? Is this supposed to modify passed in variable? If not I think it's better to leave it as a pass by value.